server/server.go

```diff
@@ -525,15 +529,75 @@ func (s *Server) loadIndex(id string) error {
 	return nil
 }
 
+const repositoryIDSize = sha256.Size
+const RepositoryVersion = 1
+
+func (s *Server) createConfig() (err error) {
+	s.Config.ChunkerPolynomial, err = chunker.RandomPolynomial()
+	if err != nil {
+		return err
+	}
+
+	newID := make([]byte, repositoryIDSize)
+	_, err = io.ReadFull(rand.Reader, newID)
+	if err != nil {
+		return err
+	}
+
+	s.Config.ID = hex.EncodeToString(newID)
+	s.Config.Version = RepositoryVersion
+
+	debug.Log("Server.createConfig", "New config: %#v", s.Config)
+
+	_, err = s.SaveJSONUnpacked(backend.Config, s.Config)
+	return err
+}
+
+func (s *Server) loadConfig(cfg *Config) error {
+	err := s.LoadJSONUnpacked(backend.Config, nil, cfg)
+	if err != nil {
+		return err
+	}
+
+	if !cfg.ChunkerPolynomial.Irreducible() {
+		return errors.New("invalid chunker polynomial")
+	}
+
+	return nil
+}
+
+// SearchKey tries to find a key for which the supplied password works,
+// afterwards the repository config is read and parsed.
 func (s *Server) SearchKey(password string) error {
 	key, err := SearchKey(s, password)
 	if err != nil {
 		return err
 	}
 
-	s.key = key
+	s.key = key.master
+	s.keyName = key.Name()
+	return s.loadConfig(&s.Config)
+}
 
-	return nil
+// CreateMasterKey creates a new key with the supplied password, afterwards the
+// repository config is created.
+func (s *Server) CreateMasterKey(password string) error {
+	has, err := s.Test(backend.Config, "")
+	if err != nil {
+		return err
+	}
+	if has {
+		return errors.New("repository master key and config already initialized")
+	}
+
+	key, err := createMasterKey(s, password)
+	if err != nil {
+		return err
+	}
+
+	s.key = key.master
+	s.keyName = key.Name()
+	return s.createConfig()
```

Comment:
I would make this three methods then. CreateMasterKey, CreateConfig, and something like Init (which calls the other two).